src/convex/ConvexAvatarMultiToken.sol

```diff
@@ -327,14 +328,22 @@ contract ConvexAvatarMultiToken is BaseAvatar, ConvexAvatarUtils, PausableUpgrad
         }
 
         IStakingProxy proxy = IStakingProxy(vaultAddr);
-        uint256 lockedBal = IFraxUnifiedFarm(proxy.stakingAddress()).lockedLiquidityOf(vaultAddr);
+        IFraxUnifiedFarm unifiedFarm = IFraxUnifiedFarm(proxy.stakingAddress());
+
+        uint256 lockedBal = unifiedFarm.lockedLiquidityOf(vaultAddr);
         if (lockedBal == 0) {
             revert NothingToWithdraw();
         }
 
         bytes32 kek = kekIds[vaultAddr];
 
-        // TODO: double-check that indeed that kek_id is expired
+        /// @dev: only inspecting index `0` given that we only use an unique lock id per private vault
+        (,,, uint256 endingTimestamp,) = unifiedFarm.lockedStakes(vaultAddr, 0);
```

Comment:
is this sufficient in case a second deposit is initiated after a kek_id has expired and been replaced with a new deposit? ie
- deposit (kek_id1)
- withdraw after lock expires
- deposit again (new id, kek_id2, no?)
- check if expired lock will now check `endingTimestamp` of kek_id1 no?

Reply:
can confirm by asserting `kekId == unifiedFarm.lockedStakes(vaultAddr, 0)[0]`

Reply:
my understanding is that when the 2nd action happens wd after lock expires is that the kek_id also within the unified farm contract will be delated from their storage, [ref](https://etherscan.io/address/0x5a92ef27f4baa7c766aee6d751f754ebdebd9fae#code#L2137), so then we can deposit again, it will create lock in the index 0, since the avatar will not handle two locks in tandem, i think checking for zero index seems unless i am missing something